plugins/org.eclipse.n4js/src/org/eclipse/n4js/utils/NodeModulesDiscoveryHelper.java

```diff
@@ -42,14 +44,25 @@
 	/** Defines whether a node_modules folder is a child of an NPM project or a YARN workspace project */
 	public static class NodeModulesFolder {
 		/** Location of a node_modules folder */
-		final public File nodeModulesFolder;
-		/** True iff the {@link #nodeModulesFolder} is in a yarn workspace */
-		final public boolean isYarnWorkspace;
+		final public File npmNodeModulesFolder;
+		/** Location of a node_modules folder */
+		final public File yarnNodeModulesFolder;
```

Comment:
I don't understand the difference between `npmNodeModulesFolder` and `yarnNodeModulesFolder` (also the identical API doc does not help 😄 ). Yarn is compatible with npm, so all node_modules folders used by npm are also used by yarn (though not vice versa). What about `localNodeModulesFolder` (for those in each project) and `globalNodeModulesFolder` (for the one in the root folder of the yarn workspace)?